regression-tests.auth-py/test_ALIAS.py

```diff
@@ -0,0 +1,201 @@
+    _zones = {
+        'example.org': """
+example.org.                 3600 IN SOA  {soa}
+example.org.                 3600 IN NS   ns1.example.org.
+example.org.                 3600 IN NS   ns2.example.org.
+ns1.example.org.             3600 IN A    {prefix}.10
+ns2.example.org.             3600 IN A    {prefix}.11
+
+noerror.example.org.         3600 IN ALIAS noerror.example.com.
+nxd.example.org.             3600 IN ALIAS nxd.example.com.
+servfail.example.org.        3600 IN ALIAS servfail.example.com
+        """,
+    }
+
+    @classmethod
+    def startResponders(cls):
+        global aliasUDPReactorRunning
+
+        address = cls._PREFIX + '.1'
+        port = 5301
+
+        if not aliasUDPReactorRunning:
+            reactor.listenUDP(port, AliasUDPResponder(), interface=address)
+
+            aliasUDPReactorRunning = True
+
+        if not reactor.running:
+            cls._ALIASResponder = threading.Thread(name='ALIAS Responder',
+                                                   target=reactor.run,
+                                                   args=(False,))
+            cls._ALIASResponder.setDaemon(True)
+            cls._ALIASResponder.start()
+
+    def testNoError(self):
+        expected_a = [dns.rrset.from_text('noerror.example.org.',
+                                          0, dns.rdataclass.IN, 'A',
+                                          '192.0.2.1')]
+        expected_aaaa = [dns.rrset.from_text('noerror.example.org.',
+                                             0, dns.rdataclass.IN, 'AAAA',
+                                             '2001:DB8::1')]
+
+        query = dns.message.make_query('noerror.example.org', 'A')
+        res = self.sendUDPQuery(query)
+        self.assertRcodeEqual(res, dns.rcode.NOERROR)
+        self.assertAnyRRsetInAnswer(res, expected_a)
+
+        query = dns.message.make_query('noerror.example.org', 'AAAA')
+        res = self.sendUDPQuery(query)
+        self.assertRcodeEqual(res, dns.rcode.NOERROR)
+        self.assertAnyRRsetInAnswer(res, expected_aaaa)
+
+        query = dns.message.make_query('noerror.example.org', 'ANY')
+        res = self.sendUDPQuery(query)
+        self.assertRcodeEqual(res, dns.rcode.NOERROR)
+        self.assertAnyRRsetInAnswer(res, expected_a)
+        self.assertAnyRRsetInAnswer(res, expected_aaaa)
+
+        # NODATA
+        query = dns.message.make_query('noerror.example.org', 'MX')
+        res = self.sendUDPQuery(query)
+        self.assertRcodeEqual(res, dns.rcode.NOERROR)
+        self.assertEqual(len(res.answer), 0)
+
+    def testNxDomain(self):
+        query = dns.message.make_query('nxd.example.org', 'A')
+        res = self.sendUDPQuery(query)
+        self.assertRcodeEqual(res, dns.rcode.SERVFAIL)
+
+        query = dns.message.make_query('nxd.example.org', 'AAAA')
+        res = self.sendUDPQuery(query)
+        self.assertRcodeEqual(res, dns.rcode.SERVFAIL)
+
+        # TODO this should actually return SOA + NS?
```

Comment:
No - ANY includes A/AAAA and thus ALIAS and thus I think SERVFAIL is fine.

Reply:
ANY is not ALL though....

Reply:
> ANY is not ALL though....

Yes it is. Cloudflare can go ** a ** :)